### PR TITLE
feat: Support UNKNOWN type for Spark collect_set aggregate function

### PR DESCRIPTION
**Pull request by WangGuangxin**
Gluten maps Spark NullType as UNKNOWN type in Velox. This PR supports the 
UNKNOWN type for the Spark collect_set aggregate function.

### CODE REVIEW

**Comment by WangGuangxin** [comment]
@zhli1142015 @rui-mo Could you please help review this?

File: velox/functions/sparksql/aggregates/CollectSetAggregate.cpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <type/Type.h>
```
Comment (zhli1142015) [inline velox/functions/sparksql/aggregates/CollectSetAggregate.cpp:16]: I think this line is unnecessary. Maybe remove it.

Comment (WangGuangxin) [reply velox/functions/sparksql/aggregates/CollectSetAggregate.cpp]: yes, removed

**Review by PHILO-HE — APPROVED** [review]

File: velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp [hunk 2]
```diff
@@ -260,5 +260,18 @@ TEST_F(CollectSetAggregateTest, rowWithNestedNull) {
...
+  auto data = makeRowVector({
+    makeNullConstant(TypeKind::UNKNOWN, 3),
+  });
+
+  auto expected = makeRowVector({
+      makeArrayVectorFromJson<int32_t>({"[]"}),
+  });
+
+  testAggregations(
+      {data}, {}, {"collect_set(c0)"}, {}, {expected});
```
Comment (rui-mo) [inline velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp:273]: Would you add some tests for null type being the grouping key?

**Review by rui-mo — COMMENTED** [review]
Thanks.

File: velox/functions/sparksql/aggregates/CollectSetAggregate.cpp [hunk 3]
```diff
@@ -90,6 +90,8 @@ void registerCollectSetAggAggregate(
             // false.
             return std::make_unique<SparkSetAggAggregate<ComplexType>>(
                 resultType, false);
+          case TypeKind::UNKNOWN:
+            return std::make_unique<SparkSetAggAggregate<UnknownValue>>(resultType);
```
[the remark this answers is not in the document]
Comment (WangGuangxin) [reply velox/functions/sparksql/aggregates/CollectSetAggregate.cpp]: fixed

File: velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp (hunk 2 above)
Comment (WangGuangxin) [reply velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp]: add one more test

File: velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp [hunk 4]
```diff
@@ -260,5 +260,23 @@ TEST_F(CollectSetAggregateTest, rowWithNestedNull) {
...
+  auto data = makeRowVector({
+      makeNullConstant(TypeKind::UNKNOWN, 3),
+  });
+
+  auto expected = makeRowVector({
+      makeArrayVectorFromJson<int32_t>({"[]"}),
+  });
+  testAggregations({data}, {}, {"collect_set(c0)"}, {}, {expected});
+
+  // with groupby key
```
Comment (rui-mo) [inline velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp:273]: nit: capitalize the first letter and add a dot at the end. Comment suggestion:

> The grouping key is of UNKONWN type.

File: velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp [hunk 5]
```diff
@@ -260,5 +260,23 @@ TEST_F(CollectSetAggregateTest, rowWithNestedNull) {
...
+  auto expected2 = makeRowVector({
```
Comment (rui-mo) [inline velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp:274]: nit: reuse 'expected'.

**Comment by PHILO-HE** [comment]
@WangGuangxin, please revise the title to replace "null type" with "UNKNOWN type", since null type is not belonging to velox.

**Comment by WangGuangxin** [comment]
> @WangGuangxin, please revise the title to replace "null type" with "UNKNOWN type", since null type is not belonging to velox.

Thanks, fixed

File: velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp [hunk 6]
```diff
@@ -260,5 +260,23 @@ TEST_F(CollectSetAggregateTest, rowWithNestedNull) {
       {data}, {}, {"collect_set(c0)"}, {"spark_array_sort(a0)"}, {expected});
 }
 
+TEST_F(CollectSetAggregateTest, nullType) {
```
Comment (rui-mo) [inline velox/functions/sparksql/aggregates/tests/CollectSetAggregateTest.cpp:263]: nit: nullType->unknownType

**Review by zhli1142015 — APPROVED** [review]
LGTM, Thanks.

**Comment by rui-mo** [comment]
@WangGuangxin Could you rebase this PR to get rid of the fuzzer failure?

**Comment by WangGuangxin** [comment]
> @WangGuangxin Could you rebase this PR to get rid of the fuzzer failure?

done

**Review by rui-mo — COMMENTED** [review]
There are some conflicts, and please help resolve them. Thanks.

**Comment by WangGuangxin** [comment]
@rui-mo rebased

**Review by rui-mo — APPROVED** [review]
